overlord/managers_test.go

```diff
@@ -12281,3 +12281,36 @@ volumes:
 	c.Assert(err, IsNil)
 	c.Check(chg.Err(), ErrorMatches, "cannot perform the following tasks:\n.*Mount snap \"pi-kernel\" \\(2\\) \\(cannot refresh kernel with change created by old snapd that is missing gadget update task\\)")
 }
+
+func (s *mgrsSuite) TestDownloadSimple(c *C) {
```

Comment:
Still need to add some tests for:
- [ ] downloading a snap with prerequisite snaps
- [ ] validate that needed assertions are downloaded
- [ ] downloading a snap at a specific revision

Probably some other things too, let me know.

Reply:
we need this test to check the validate-snap part of this? as that task comes from assertstate?

Reply:
I've added a check to make sure that the snap revision assertion is added to the database.